### ISSUE TITLE
animation hang since fedora37 upgrade

### ISSUE THREAD

**Issue by emmanueltouzery** [neovide/neovide#1700]
**Describe the bug**
Since I upgraded to fedora 37 (from fedora 36) I get quite frequently hangs mid-animation, especially (maybe only?) in telescope. So the animation stops while it's not finished. If I type any key or move the mouse, the display updates immediately to finish the animation.

**To Reproduce**
1. Start the application on fedora37 with multigrid (might be specific to amdgpu which I'm using?)
2. Open telescope find_files repeatedly until you reproduce the "hang"
3. Type in key or move the mouse to make neovide update the GUI again

**Expected behavior**
The GUI shouldn't "hang" as it does from time to time.

**Screenshots**
Crop of the picture. The application stays indefinitely with this display until I move the mouse or type on the keyboard.
![image](https://user-images.githubusercontent.com/339433/209577861-97112eda-b178-42a1-bdf7-f74f9837128e.png)

**Desktop (please complete the following information):**
 - OS: fedora37
 - Neovide Version 67a48653b71bf8a7c36953ddd75dc06894628d9a
 - Neovim Versionv0.8.0-1210-gd367ed9b2

**Please run `neovide --log` and paste the contents of the `.log` file created in the current directory here:**
[neovide_rCURRENT.log](https://github.com/neovide/neovide/files/10304581/neovide_rCURRENT.log)

**Additional context**
-

**Comment by emmanueltouzery** [comment]
important information: I have not been able to reproduce the issue when running neovide with the `--noidle` flag.

I'll keep running neovim with the `--noidle` flag and will report if I get the issue also with that flag enabled.

**Comment by emmanueltouzery** [comment]
sadly i got the issue also with `--noidle` now.. but it is less frequent with `--noidle`.

**Comment by emmanueltouzery** [comment]
I really don't know what I'm doing but for now I'm running it with this change, and I didn't reproduce the issue again:
```diff
diff --git a/src/window/mod.rs b/src/window/mod.rs
index 82aa281..0028d9f 100644
--- a/src/window/mod.rs
+++ b/src/window/mod.rs
@@ -194,6 +194,7 @@ impl GlutinWindowWrapper {
     }
 
     pub fn draw_frame(&mut self, dt: f32) {
+        trace!("draw_frame");
         let window = self.windowed_context.window();
         let new_size = window.inner_size();
 
@@ -509,7 +510,8 @@ pub fn create_window() {
         let expected_frame_length_seconds = 1.0 / refresh_rate;
         let frame_duration = Duration::from_secs_f32(expected_frame_length_seconds);
 
-        if frame_start - previous_frame_start > frame_duration {
+        if true {
+            // if frame_start - previous_frame_start > frame_duration {
             let dt = previous_frame_start.elapsed().as_secs_f32();
             window_wrapper.draw_frame(dt);
             if let FocusedState::UnfocusedNotDrawn = focused {
```
I noticed that when the issue is present, in the trace logs, it talks about queuing a frame, but "draw_frame" is not printed. It may be that this doesn't help at all, it's hard to tell with issues that don't happen deterministically. I'll report, but I'm using my editor less this week, since I'm on holidays.

**Comment by emmanueltouzery** [comment]
so my change with `if true` was just affecting too many things. it did seem to fix the hang, but made eg mouse scrolling quite bad.
i looked more at the trace output... as i mentioned in my previous comment "in the trace logs, it talks about queuing a frame, but "draw_frame" is not printed".

what I found is that when there is such a "hang" it seems the control flow gets stuck within `draw_frame`. I put more trace entries and looked where the execution stops when i see a "hang".

it stops right there:
```rust
            self.skia_renderer.gr_context.flush(None);
            trace!("d3");
            self.windowed_context.swap_buffers().unwrap();
            trace!("d4");
```

d3 is printed, but d4 is not.

And I noticed this comment:
```rust
    /// Swaps the buffers in case of double or triple buffering.
    ///
    /// You should call this function every time you have finished rendering, or
    /// the image may not be displayed on the screen.
    ///
    /// **Warning**: if you enabled vsync, this function will block until the
    /// next time the screen is refreshed. However drivers can choose to
    /// override your vsync settings, which means that you can't know in
    /// advance whether `swap_buffers` will block or not.
    pub fn swap_buffers(&self) -> Result<(), ContextError> {
        self.context.context.swap_buffers()
    }
```

however even disabling vsync with the command-line flag `--novsync` I still get the issue.

**Comment by emmanueltouzery** [comment]
...and my latest experiment, for now I believe that `WINIT_UNIX_BACKEND=x11` is a workaround that prevents the issue from occuring. I'll report if it turns out the issue still appears despite that.

**Comment by emmanueltouzery** [comment]
This is fixed by with 0.10.4.
